learning_assistant/utils.py

```diff
@@ -0,0 +1,42 @@
+"""
+Utils file for learning-assistant.
+"""
+import logging
+
+import requests
+from django.conf import settings
+from requests.exceptions import ConnectTimeout
+from rest_framework import status as http_status
+
+log = logging.getLogger(__name__)
+
+
+def get_chat_response(message_list):
+    """
+    Pass message list to chat endpoint, as defined by the CHAT_COMPLETION_API setting.
+    """
+    completion_endpoint = getattr(settings, 'CHAT_COMPLETION_API', None)
+    if completion_endpoint:
+        headers = {'Content-Type': 'application/json'}
+        connect_timeout = getattr(settings, 'CHAT_COMPLETION_API_CONNECT_TIMEOUT', 1)
+        read_timeout = getattr(settings, 'CHAT_COMPLETION_API_READ_TIMEOUT', 10)
+
+        try:
+            response = requests.post(
+                completion_endpoint,
+                headers=headers,
+                data=message_list,
+                timeout=(connect_timeout, read_timeout)
+            )
+            chat = response.json()
+            response_status = response.status_code
+        except (ConnectTimeout, ConnectionError) as e:
+            error_message = str(e)
+            log.error('Failed to connect to chat completion API. %(error)s', {'error': error_message})
+            chat = error_message
+            response_status = http_status.HTTP_502_BAD_GATEWAY
+    else:
+        response_status = http_status.HTTP_404_NOT_FOUND
```

Comment:
I'm not sure if `404` is the most appropriate HTTP status here, as it implies a client-side error, and it's an issue of server-side configuration. Maybe something from the `500` class would be more appropriate?

Reply:
I'm not seeing anything in the 500s that would really describe this situation. The reason I chose 404 is because there's no resource associated with the endpoint even though the endpoint itself is valid. This seemed like the most appropriate status code for this case.

Reply:
I thought 404 is referring to the requested resource (i.e. the chatbot feature for this course, in this case) not existing. Like, if the waffle flag for the course was disabled, I'd expect a 400 class error. The 400 class of errors is telling the client it did something wrong (e.g. requested a page that doesn't exist or an object or resource that doesn't exist), whereas the reality is that the server is misconfigured in this case. It doesn't seem like a client issue to me. A request to this endpoint is not going to work for any request to this endpoint if that setting is missing. Maybe a `501 Not Implemented` would make sense, but I can't really tell if that's referring purely to the request method not being supported or _something_ in the endpoint not being supported, which would be closer to this case.

A 404 is fine with me. I'm not an expert on status codes, so I don't want to get hung up on it!